modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/graphql/dto/v1_0/ObjectDefinitionGraphQLDTOContributor.java

```diff
@@ -260,17 +260,43 @@ public <T> T getRelationshipValue(
 		ObjectEntry objectEntry = _objectEntryManager.getObjectEntry(
 			dtoConverterContext, _objectDefinition, id);
 
-		long relationshipId = _getRelationshipId(objectEntry.getProperties());
+		Map<String, Object> properties = objectEntry.getProperties();
+
+		long relationshipId = _getRelationshipId(properties);
 
 		if (relationshipId <= 0) {
-			Page<ObjectEntry> page =
-				_objectEntryManager.getObjectEntryRelatedObjectEntries(
-					dtoConverterContext, _objectDefinition, id,
-					relationshipName,
-					Pagination.of(QueryUtil.ALL_POS, QueryUtil.ALL_POS));
-
-			return (T)TransformUtil.transform(
-				page.getItems(), itemObjectEntry -> _toMap(itemObjectEntry));
+			String objectField2Name = _getObjectField2Name(
+				properties, relationshipName);
+
+			if (objectField2Name == null) {
```

Comment:
could you please replace this condition with `if (objectField2Name == null || relationshipId <= 0)`